spec.html

```diff
@@ -142,6 +142,18 @@ <h1>Updated Syntactic Grammar Productions</h1>
         CallExpression[?Yield, ?Await] `.` IdentifierName
         CallExpression[?Yield, ?Await] TemplateLiteral[?Yield, ?Await]
         <ins>CallExpression[?Yield, ?Await] `.` PrivateIdentifier</ins>
+
+      OptionalChain[Yield, Await] :
+        `?.` `[` Expression[+In, ?Yield, ?Await] `]`
+        `?.` IdentifierName
+        `?.` Arguments[?Yield, ?Await]
+        `?.` TemplateLiteral[?Yield, ?Await, +Tagged]
+        <ins>`?.` PrivateIdentifier</ins>
```

Comment:
Wasn't the idea that this production would be omitted?

Reply:
See line 198, it’s an explicit syntax error (like the optional template literal syntax, eg)

Reply:
Ah, I see. I guess I find this editorial strategy strange--isn't just about everything that's not in the grammar reserved in the future? We haven't used this editorial strategy for other things that we've been thinking about including in the future, like private fields in object literals, or concise private field references (without `this`). I don't know why this should be treated differently.

Reply:
I left it this way because the only 3 implementations that support privates and optional chain didn't realize that mixing them isn't in the spec. Conveniently none of them screwed up this production, but maybe the next implementation would.

Reply:
Good point; we should be clear here. How about a note explaining that this production is specifically omitted?

Reply:
Isn’t that what’s on line 198?

Reply:
Are you asking for an emu-note here? If not, I think the Early Error note already accomplishes this? https://github.com/tc39/proposal-class-fields/pull/301/files#diff-3540caefa502006d8a33cb1385720803R223-R227

Reply:
Right, I'm suggesting to remove the grammatical production and early error, and instead simply have a note that explains that the grammatical production is deliberately excluded.

Reply:
FYI, there are Test262 cases for this https://github.com/tc39/test262/pull/2408. We also considered this issue while implementing private features on WebKit. Unfortunately, I don't remember who report this issue.

Reply:
@caiolima Are there any tests for the other case, `x?.y.#z`? (I think we should land this patch, which permits this grammar, but currently, it's not permitted in the grammar.)

Reply:
Done.

Reply:
> @caiolima Are there any tests for the other case, `x?.y.#z`? (I think we should land this patch, which permits this grammar, but currently, it's not permitted in the grammar.)

I don't think we have.